### PR TITLE
Added Color changing QLabel for Packet loss indication

### PR DESCRIPTION
**Pull request by sauravbanna**
### Description

We want some sort of visual indicator for when we have packet and world loss during a game. So this PR adds text labels that become more red as the packet loss percent increases, up till a max value.

https://github.com/UBC-Thunderbots/Software/assets/90598576/a2795a4b-c18a-4d53-91f4-bace8b2dc1ec

### Testing Done

Thunderscope builds and runs normally. Color changing works correctly for the correct values when testing with fake messages.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/thunderscope/common/common_widgets.py [hunk 1]
```diff
@@ -88,27 +130,32 @@ def __init__(self, min_val, max_val, decimals=2):
...
         percent = min(
             1,
-            max(0, (self.value() - self.minimum()) / (self.maximum() - self.minimum())),
+            max(
+                0,
+                int(
+                    (self.value() - self.minimum()) / (self.maximum() - self.minimum())
+                ),
+            ),
         )
```
Comment (itsarune) [inline src/software/thunderscope/common/common_widgets.py]: nit: you could make this a helper function

**Review by itsarune — COMMENTED** [review]
looks good! For these types of PRs, I suggest attaching a gif of Thunderscope with your new change.

File: src/proto/robot_status_msg.proto
Comment (nimazareian) [inline src/proto/robot_status_msg.proto:1]: This file and anything related to World (including packet loss should be removed! Seems like a bad merge.

File: src/software/thunderscope/constants.py [hunk 2]
```diff
@@ -81,6 +81,10 @@ class EstopMode(IntEnum):
     PHYSICAL_ESTOP = 2
 
 
+# the maximum packet / world loss percent indicated by UI
+MAX_ACCEPTABLE_LOSS_PERCENT = 30
```
Comment (nimazareian) [inline src/software/thunderscope/constants.py]:
```suggestion
MAX_ACCEPTABLE_PACKET_LOSS_PERCENT = 30
```

File: src/software/thunderscope/robot_diagnostics/robot_view.py
Comment (nimazareian) [inline src/software/thunderscope/robot_diagnostics/robot_view.py:1]: I don't think you meant to commit this

**Review by nimazareian — APPROVED** [review]
LGTM

**Review by itsarune — APPROVED** [review]
LGTM

**Pull request merged**